### PR TITLE
implement users-search store configuration, add typescript import aliases

### PR DESCRIPTION
**Pull request by tarvyn**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,10 +1,22 @@
 import { NgModule } from '@angular/core';
-import { Routes, RouterModule } from '@angular/router';
+import { RouterModule, Routes } from '@angular/router';
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: '',
+    redirectTo: '/users-search',
```
Comment (danemesis) [inline src/app/app-routing.module.ts:7]: what the idea then here to use lazy loading ?

File: src/app/features/users-search/users-search-container.component.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<span>Users search:</span>
+<input type="text">
+
+<div class="search-results" *ngIf="users$ | async; let users">
+  <div *ngFor="let user of users">{{user.id}}</div>
```
Comment (danemesis) [inline src/app/features/users-search/users-search-container.component.html]: trackBy?

Comment (tarvyn) [reply src/app/features/users-search/users-search-container.component.html]: Totally agree) thanks

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (tarvyn) [reply src/app/app-routing.module.ts:7]: For consistency. I guess each of three feature modules (Users, ReposList, RepoInfo) should be lazy-loaded.

Comment (danemesis) [reply src/app/app-routing.module.ts:7]: got you. just to be aware, that, if I remember right, that lazy loaded modules have their own DI, so if we rely on services used across, we have to take into account this. Sorry, haven't had time to look more precisely, but look good, great job 👍

File: src/app/api-connectors/github-api-connector.service.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  {
+    id: 1
+  }
+];
+
+// TODO: replace mocks with api call
+@Injectable({
+  providedIn: 'root'
+})
+export class GithubApiConnectorService {
```
Comment (OlegMal88) [inline src/app/api-connectors/github-api-connector.service.ts]: all exports in the end of the page

File: src/app/features/users-search/users-search-container.component.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<span>Users search:</span>
+<input type="text">
+
+<div class="search-results" *ngIf="users$ | async; let users">
```
Comment (OlegMal88) [inline src/app/features/users-search/users-search-container.component.html]: ngIf from the next line

File: src/app/features/users-search/users-search-container.component.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { select, Store } from '@ngrx/store';
+import * as fromUsers from '@store/users-store';
+
+@Component({
+  selector: 'app-users-search-container',
+  templateUrl: './users-search-container.component.html',
+  styleUrls: ['./users-search-container.component.scss']
+})
+export class UsersSearchContainerComponent {
+  users$ = this.store.pipe(select(fromUsers.selectAllUsers));
```
Comment (OlegMal88) [inline src/app/features/users-search/users-search-container.component.ts:12]: uncovered by unit tests

File: src/app/store/users-store/effects.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import * as usersActions from './actions';
+
+@Injectable()
+export class UsersStoreEffects {
+  @Effect()
+  loadRequestEffect$: Observable<Action> = this.actions$.pipe(
+    ofType<usersActions.SearchUsersStartAction>(
+      usersActions.ActionTypes.SEARCH_USERS_START
+    ),
+    startWith(new usersActions.SearchUsersStartAction({ search: '' })),
```
Comment (OlegMal88) [inline src/app/store/users-store/effects.ts:17]: ?

File: src/app/store/users-store/effects.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    switchMap(action =>
```
Comment (OlegMal88) [inline src/app/store/users-store/effects.ts:18]: type of Action

File: src/app/store/users-store/effects.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      this.githubApiConnectorService
+        .searchUsers(action.payload.search)
+        .pipe(
+          map(users => new usersActions.SearchUsersSuccessAction({ users })),
```
Comment (OlegMal88) [inline src/app/store/users-store/effects.ts:22]: users and error type

File: src/app/store/users-store/selectors.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+export const selectUserById = (id: GithubUser['id']) =>
+  createSelector(selectAllUsers, (users: Array<GithubUser>) => {
+    if (users) {
+      return users.find(user => user.id === id);
+    }
+    return null;
+  });
+
+export const selectUsersSearchError: MemoizedSelector<object, string> = createSelector(
```
Comment (OlegMal88) [inline src/app/store/users-store/selectors.ts]: why object?

File: src/app/store/users-store/users-store.module.ts [hunk 10]
```diff
@@ -2,16 +2,15 @@ import { CommonModule } from '@angular/common';
...
-import { EntityFeatureStoreEffects } from './effects';
+import { UsersStoreEffects } from './effects';
 import { featureReducer } from './reducer';
 
 @NgModule({
   imports: [
     CommonModule,
-    StoreModule.forFeature('entityFeature', featureReducer),
-    EffectsModule.forFeature([EntityFeatureStoreEffects])
+    StoreModule.forFeature('users', featureReducer),
```
Comment (OlegMal88) [inline src/app/store/users-store/users-store.module.ts]: put 'users' into const. NO magic strings

**Review by OlegMal88 — CHANGES_REQUESTED** [review]

File: src/app/store/users-store/effects.ts (hunk 7 above)
Comment (tarvyn) [reply src/app/store/users-store/effects.ts:18]: it's already inferred from 
`
    ofType<usersActions.SearchUsersStartAction>(
      usersActions.ActionTypes.SEARCH_USERS_START
    )
`

File: src/app/store/users-store/effects.ts (hunk 8 above)
Comment (tarvyn) [reply src/app/store/users-store/effects.ts:22]: users are inferred from 
`
    ofType<usersActions.SearchUsersStartAction>(
      usersActions.ActionTypes.SEARCH_USERS_START
    )
`

**Pull request merged**